src/main/scala/mesosphere/marathon/state/GroupManager.scala

```diff
@@ -138,17 +139,19 @@ class GroupManager @Inject() (
 
     log.info(s"Upgrade group id:$gid version:$version with force:$force")
 
-    def storeUpdatedApps(plan: DeploymentPlan): Future[Unit] = {
-      plan.affectedApplicationIds.foldLeft(Future.successful(())) { (savedFuture, currentId) =>
+    case class AppDelta(updated: Seq[AppDefinition], deleted: Seq[PathId])
```

Comment:
The DeploymentPlan has a function `createdOrUpdatedApps`.
Perhaps also add a `deletedApps` and use those functions instead of computing AppDelta in GroupManager?